### ISSUE TITLE
False positive for prefer_initializing_formals

### ISSUE THREAD

**Issue by tijoforyou** [dart-lang/linter#2441]
See this example code
```dart
class Foo {
  String value;

  Foo();

  factory Foo.withValue(String value) {
    var foo = Foo();
    foo.value = value;  // <-- prefer_initializing_formals false positive
    return foo;
  }
}
```

The lint report is incorrect here.
